content/ru/docs/concepts/overview/working-with-objects/finalizers.md

```diff
@@ -0,0 +1,53 @@
+---
+title: Финализаторы
+content_type: concept
+weight: 80
+---
+
+# Финализаторы
+
+Финализаторы — это ключи в пространстве имен, которые сообщают Kubernetes дождаться выполнения определенных условий, прежде чем он полностью удалит ресурсы, помеченные для удаления. Финализаторы предупреждают контроллеры о необходимости очистить ресурсы, принадлежащие удаленному объекту.
+
+Когда вы указываете Kubernetes удалить объект, для которого указаны финализаторы, API Kubernetes помечает объект для удаления, заполняя .metadata.deletionTimestamp, и возвращает код состояния 202 (HTTP «Принято»). Целевой объект остается в завершающем состоянии, пока плоскость управления или другие компоненты выполняют действия, определенные финализаторами. После завершения этих действий контроллер удаляет соответствующие финализаторы из целевого объекта. Если поле Metadata.finalizers пусто, Kubernetes считает удаление завершенным и удаляет объект.
+
+Вы можете использовать финализаторы для управления сборкой мусора ресурсов. Например, вы можете определить финализатор для очистки связанных ресурсов или инфраструктуры перед тем, как контроллер удалит целевой ресурс.
+
+Вы можете использовать финализаторы для управления сборкой мусора объектов, предупреждая контроллеры о необходимости выполнения определенных задач очистки перед удалением целевого ресурса.
+
+Финализаторы обычно не указывают код для выполнения. Вместо этого они обычно представляют собой списки ключей определенного ресурса, похожие на аннотации. Kubernetes автоматически определяет некоторые финализаторы, но вы также можете указать свои собственные.
+
+## Как работают финализаторы
+
+Когда вы создаете ресурс с помощью файла манифеста, вы можете указать финализаторы в поле Metadata.finalizers. Когда вы пытаетесь удалить ресурс, сервер API, обрабатывающий запрос на удаление, замечает значения в поле финализаторов и выполняет следующее:
+
+* Изменяет объект, добавляя поле метаданных.deletionTimestamp со временем начала удаления.
+* Предотвращает удаление объекта до тех пор, пока все элементы не будут удалены из его поля метаданных.finalizers.
+* Возвращает код состояния 202 (HTTP «Принято»)
+
+Контроллер, управляющий этим финализатором, замечает обновление объекта, устанавливающее метаданные.deletionTimestamp, что указывает на то, что было запрошено удаление объекта. Затем контроллер пытается удовлетворить требования финализаторов, указанных для этого ресурса. Каждый раз, когда условие финализатора удовлетворяется, контроллер удаляет этот ключ из поля финализаторов ресурса. Когда поле финализаторов очищается, объект с установленным полем deleteTimestamp автоматически удаляется. Вы также можете использовать финализаторы, чтобы предотвратить удаление неуправляемых ресурсов.
```

Comment:
The first sentence here seems difficult to read. For example, we can split it into two sentences.